distributed/tests/test_utils_test.py

```diff
@@ -606,3 +607,33 @@ def test_start_failure_scheduler():
     with pytest.raises(TypeError):
         with cluster(scheduler_kwargs={"foo": "bar"}):
             return
+
+
+def test_invalid_transitions(capsys):
+    @gen_cluster(client=True, nthreads=[("127.0.0.1", 1)])
+    async def test_log_invalid_transitions(c, s, a):
+        x = c.submit(inc, 1, key="task-name")
+        y = c.submit(inc, x)
+        xkey = x.key
+        del x
+        await y
+        while a.tasks[xkey].state != "released":
+            await asyncio.sleep(0.01)
+        ts = a.tasks[xkey]
+        with pytest.raises(InvalidTransition):
+            a.transition(ts, "foo", stimulus_id="bar")
+
+        while not s.events["invalid-worker-transition"]:
+            await asyncio.sleep(0.01)
+
+    with pytest.raises(Exception) as info:
```

Comment:
```suggestion
    with pytest.raises(ValueError) as info:
```

Reply:
Can you motivate this change?  Do you care in particular that it is a ValueError, or are you just trying to match the behavior as it exists today?

Reply:
And, to make my thinking explicit here, I don't care if this is a ValueError or some other kind of error.  I personally like using ValueErrors but sometimes folks here like to use very specific things.  If they did that in the future I wouldn't want to hold them back.  I *would* want to hold them back if relevant data wasn't in the error message (as is enforced below).

I've chosen to use Exception here because I don't have a strong enough preference on type to want to constrain future developers too much.

Reply:
> Can you motivate this change?

Just for the normal reason of catching the specific exception type, instead of any `Exception`: don't accidentally catch something else. I saw that `check_invalid_worker_transitions` raises `ValueError` currently, which is what you're hoping to catch.

If, say, the cluster actually deadlocked and timed out, the `assert "invalid" in str(info).lower()` _should_ fail the test anyway, so it's not that big of a deal. It's just a little easier to read the intention of the test this way.

> I've chosen to use Exception here because I don't have a strong enough preference on type to want to constrain future developers too much

If future developers want to change the exception type, then can't they can make a one-line change to the test?

Reply:
> If, say, the cluster actually deadlocked and timed out, the assert "invalid" in str(info).lower() should fail the test anyway, so it's not that big of a deal. It's just a little easier to read the intention of the test this way.

Yeah, I don't think that we're going to catch unwanted things here.  I would also want to be way more specific in actual code.

> If future developers want to change the exception type, then can't they can make a one-line change to the test?

Yeah, in this case it's pretty cheap.  In general though I like to test only things that I care about.  Otherwise we end up asking devs to change all sorts of unrelated tests that they don't know anything about, which error prone.  In general my practice here is to "test things you care about, don't test to match current behavior" which is something that I'd love to socialize a bit across the project (historically we didn't do this, and it resulted in significant pain).  In this case though I can't come up with any way to trigger this test except in that one line, so I don't care that much.  I'm just sticking to my default policy.